### ISSUE TITLE
Accuracy mismatches between Table 1 and Figure 3/4 in FLAN XL 2022

### ISSUE THREAD

**Issue by jianguoz** [google-research/FLAN#33]
Thanks for open-sourced the FLAN collections. We noticed that in the paper [The Flan Collection: Designing Data and Methods for Effective Instruction Tuning](https://arxiv.org/pdf/2301.13688.pdf). Table 1 shows that the 0-shot and 5-shot accuracy of `TX-XL Flan 2022`  on MMLU are **50.3/52.4**. 
However, in both Figure 3 and Figure 4, the highest 0-shot and 5-shot accuracy achieved by `TX-XL Flan 2022` is below **46.0**. It is unclear which accuracy numbers are correct.

Additionally,  Table 5 of [Scaling Instruction-Finetuned Language Models](https://arxiv.org/pdf/2210.11416.pdf). The 5-shot accuracy of `T5-XL` is **52.4**, which is the same as `TX-XL Flan 2022`. Are they use same data settings or by chance the numbers are same?

We may missed information here. Can we know which numbers should be correct? Thanks:)

**Comment by shayne-longpre** [comment]
@jianguoz: Thank you for your question.

The ablation experiments in Figures 3 and 4 are running on simplified settings, before we'd fully tuned and combined everything to get the best models presented in Table 1 (the same one from Table 5 in "[Scaling Instruction-Finetuned Language Models](https://arxiv.org/pdf/2210.11416.pdf)"). If I remember correctly, Figures 3 is running only on the Flan 2021 data, rather than all the collections together, and Figure 4 is just zero-shot training, not zero-shot + few-shot together. It would have been very expensive to combine all these experiments again after knowing the best settings from each. I will update the paper in the near future with these clarifications in the appendix.

**Comment by jianguoz** [comment]
Hi @shayne-longpre, thanks so much for your clarification!

**Comment by jianguoz** [comment]
Hi @shayne-longpre, we are trying to reproduce the results on MMLU. Can we know that where we can found the instructions you used to evaluate mmlu? Thanks:)

**Comment by shayne-longpre** [comment]
@jianguoz we follow the standard instructions to use the MMLU benchmark. [Here](https://github.com/jasonwei20/flan-2) is an example of the prompts we used for the chain-of-thought version. You can remove the explanation from the few-shot answers and it will be near identical to what we used for regular evaluation.

Is this what you are asking for?

**Comment by jianguoz** [comment]
Hi @shayne-longpre, thanks for your reply! I checked and run 'google/flan-t5-xl' using the standard instructions on the [MMLU benchmark](https://github.com/hendrycks/test).  The results are **48.7/49.3** for 0-shot/5-shot, which are identical to numbers reported in the MMLU benchmark, but they are much below the numbers (**50.3/52.4**) reported in the paper. Can you check this? thanks:)

**Comment by shayne-longpre** [comment]
@jianguoz Oh I see what you're saying. I have not seen the `evaluate_flan.py` script they wrote so am not sure how the prompt is different. 

Hyung Won, who did the Flan-T5 MMLU evaluations, is not at Google anymore, but I think they were definitely selected from among a few inference template options. The only other factor I can think of is the few-shot exemplars that were chosen (how many and make sure they are specific to the sub-task).

Let me know if results improve with those fixes. I can ask my internal co-authors as well.

**Comment by jianguoz** [comment]
Hi @shayne-longpre, thanks for your reply:) I check the 5-shot exemplar of mmlu-cot, which are identical to the official exemplars except that the relevant orders of some examples are slightly different.  The results does not change based on the `evaluate_flan.py`. Is that possible for you check your internal used version? Thanks a lot.

**Comment by jianguoz** [comment]
Hi @shayne-longpre, thanks for your further details! I think your points are reasonable and really makes sense!  Now closes this issue:)
